src/middlewared/middlewared/plugins/rsync.py

```diff
@@ -563,11 +563,19 @@ def run(self, job, id):
         rsync = self.middleware.call_sync('rsynctask._get_instance', id)
         commandline = self.middleware.call_sync('rsynctask.commandline', id)
 
+        for klass in ('RsyncSuccess', 'RsyncFailed'):
```

Comment:
I think you should do it after `run_command_with_user_context` so there would be no time gap between alert gone and alert appearance.